frontend/tests/playwright/product/product_option.spec.ts

```diff
@@ -1,69 +1,75 @@
+         .getByTestId('product-price')
          .first()
-         .invoke('text')
-         .as('firstOptionSkuText');
-      cy.get('@selector')
-         .get('option:selected')
-         .invoke('text')
-         .as('firstOptionName');
+         .innerText();
+      const firstOptionSku = await page
+         .getByTestId('product-sku')
+         .textContent();
+      const firstOptionName = await page
+         .getByTestId('product-option-selector')
+         .locator('option')
+         .nth(0)
+         .textContent();
 
       // Add the first product option to the cart
-      cy.findByTestId('product-add-to-cart-button').click();
-      cy.findByTestId('cart-drawer-close').click();
+      await page.getByTestId('product-add-to-cart-button').click();
+      await page.getByTestId('cart-drawer-close').click();
 
-      cy.get('@selector')
-         .select(1)
-         .then(() => {
-            cy.contains(this.firstOptionSkuText).should('not.exist');
-         });
+      // Switch to the second product option
+      await page
+         .getByTestId('product-option-selector')
+         .selectOption({ index: 1 });
+      expect(page.getByTestId('product-sku')).not.toContain(firstOptionSku);
 
       // Get the price, sku, and name for the second product option
-      cy.findAllByTestId('product-price')
-         .first()
-         .invoke('text')
-         .as('secondOptionPrice');
-      cy.findAllByTestId('product-sku')
+      const secondOptionPrice = await page
+         .getByTestId('product-price')
          .first()
-         .invoke('text')
-         .as('secondOptionSkuText');
-      cy.get('@selector')
-         .get('option:selected')
-         .invoke('text')
-         .as('secondOptionName');
+         .innerText();
+      const secondOptionSku = await page
+         .getByTestId('product-sku')
+         .textContent();
+      const secondOptionName = await page
+         .getByTestId('product-option-selector')
+         .locator('option')
+         .nth(1)
+         .textContent();
+
+      expect(firstOptionName).not.toEqual(secondOptionName);
 
       // Add the second product option to the cart
-      cy.findByTestId('product-add-to-cart-button').click();
+      await page.getByTestId('product-add-to-cart-button').click();
 
       // Assert that the cart drawer contains the skus and prices of the added products
-      cy.findByTestId('cart-drawer-body')
-         .should('be.visible')
-         .then((el) => {
-            // Parse out the skus from the skuTexts which are in form of "Item # IF145-278-14"
-            const sku1 = this.firstOptionSkuText.match(/IF\d*\-\d*\-\d*/g)[0];
-            const sku2 = this.secondOptionSkuText.match(/IF\d*\-\d*\-\d*/g)[0];
+      await expect(page.getByTestId('cart-drawer-body')).toBeVisible();
+
+      // Parse out the skus from the skuTexts which are in form of "Item # IF145-278-14"
+      const sku1 = firstOptionSku?.replace('Item # ', '') ?? '';
```

Comment:
I'm a little concerned that if it doesn't find the firstOptionSku, `sku1` will be an empty string. Then down below, we will grab an element like `getByText('');` which will probably grab anything.

I wonder if we should add an extra assertion to assert `sku1` is not an empty string.

Reply:
Addressed in [7593275](https://github.com/iFixit/react-commerce/pull/1149/commits/75932750cedac27415b6846b3991eeada354a00e)